tests/integration/test_nft_creation.py

```diff
@@ -0,0 +1,418 @@
+
+    # Consumer now should be manager
+    assert erc721_token.get_permissions(consumer.address)[0]
+
+    # Check the rest of roles for another_consumer
+    assert not erc721_token.get_permissions(another_consumer.address)[0]
+    assert not erc721_token.get_permissions(another_consumer.address)[1]
+    assert not erc721_token.get_permissions(another_consumer.address)[2]
+    assert not erc721_token.get_permissions(another_consumer.address)[3]
+
+    erc721_token.add_to_create_erc20_list(another_consumer.address, consumer)
+    erc721_token.add_to_725_store_list(another_consumer.address, consumer)
+    erc721_token.add_to_metadata_list(another_consumer.address, consumer)
+
+    # Test rest of add roles functions with newly added manager
+    assert erc721_token.get_permissions(another_consumer.address)[1]
+    assert erc721_token.get_permissions(another_consumer.address)[2]
+    assert erc721_token.get_permissions(another_consumer.address)[3]
+
+    # Remove the manager
+    erc721_token.remove_manager(consumer.address, publisher)
+
+    assert not erc721_token.get_permissions(consumer.address)[0]
+
+
+def test_properties(web3, config):
+    """Tests the events' properties."""
+
+    erc721_factory = ERC721FactoryContract(web3, get_nft_factory_address(config))
+    assert (
+        erc721_factory.event_NFTCreated.abi["name"]
+        == ERC721FactoryContract.EVENT_NFT_CREATED
+    )
+    assert (
+        erc721_factory.event_TokenCreated.abi["name"]
+        == ERC721FactoryContract.EVENT_TOKEN_CREATED
+    )
+    assert (
+        erc721_factory.event_NewPool.abi["name"] == ERC721FactoryContract.EVENT_NEW_POOL
+    )
+    assert (
+        erc721_factory.event_NewFixedRate.abi["name"]
+        == ERC721FactoryContract.EVENT_NEW_FIXED_RATE
+    )
+
+
+def test_fail_create_erc20(web3, config):
+    """Test erc20 creation fail"""
+
+    publisher = get_publisher_wallet()
+
+    erc721_factory = ERC721FactoryContract(web3, get_nft_factory_address(config))
+
+    with pytest.raises(exceptions.ContractLogicError) as err:
+        erc721_factory.create_token(
+            1,
+            ["ERC20DT1", "ERC20DT1Symbol"],
+            [publisher.address, publisher.address, publisher.address, ZERO_ADDRESS],
+            [web3.toWei("1.0", "ether"), 0],
+            [b""],
+            publisher,
+        )
+    assert (
+        err.value.args[0]
+        == "execution reverted: VM Exception while processing transaction: revert ERC721Factory: ONLY ERC721 "
+        "INSTANCE FROM ERC721FACTORY"
+    )
+
+
+def test_nonexistent_template_index(web3, config):
+    """Test erc721 non existent template creation fail"""
+
+    publisher = get_publisher_wallet()
+    erc721_factory = ERC721FactoryContract(web3, get_nft_factory_address(config))
+    with pytest.raises(exceptions.ContractLogicError) as err:
+        erc721_factory.deploy_erc721_contract(
+            "DT1",
+            "DTSYMBOL",
+            7,
```

Comment:
I don't like the usage of this `7`. What if someone creates templates and this index will suddenly be valid on their test setup? Can we instead list templates and make sure that the template index indeed does not exist? Then proceed with the test?

Reply:
@calina-c I haven't found a way to list the templates so I have done this: https://github.com/oceanprotocol/ocean.py/pull/546/files#r741006386